docs/synapse/httpapi.rst

```diff
@@ -0,0 +1,804 @@
+
+Cortex
+------
+
+A Synapse Cortex implements an HTTP API for interacting with the hypergraph and data model.  Some
+of the provided APIs are pure REST APIs for simple data model operations and single/simple node
+modification.  However, many of the HTTP APIs provided by the Cortex are streaming APIs which use
+HTTP chunked encoding to deliver a stream of results as they become available.
+
+/api/v1/feed
+~~~~~~~~~~~~
+
+The Cortex feed API endpoint allows the caller to add nodes in bulk.
+
+*Method*
+    POST
+
+    *Input*
+        The API expects the following JSON body::
+
+            {
+                "items": [ <node>, ... ],
+                # and optionally...
+                "view": <iden>,
+            }
+
+        Each ``<node>`` is expected to be in packed tuple form::
+
+            [ [<formname>, <formvalu>], {...} ]
+
+    *Returns*
+        The API returns ``{"status": "ok", "result": null}`` on success and any failures
+        are returned using the previously mentioned REST API convention.
+
+/api/v1/storm
+~~~~~~~~~~~~~
+
+The Storm API endpoint allows the caller to execute a Storm query on the Cortex and stream
+back the messages generated during the Storm runtime execution.  In addition to returning nodes,
+these messsages include events for node edits, tool console output, etc. This streaming API has back-pressure,
+and will handle streaming millions of results as the reader consumes them.
+For more information about Storm APIs, including opts behavior, see :ref:`dev_storm_api`.
+
+*Method*
+    GET
+
+    *Input*
+        The API expects the following JSON body::
+
+            {
+                "query": "a storm query here",
+
+                # optional
+                "opts": {
+                   ...
+                }
+
+                # optional 
+                "stream": "jsonlines"
+            }
+
+    *Returns*
+        The API returns a series of messages generated by the Storm runtime.  Each message is
+        returned as an HTTP chunk, allowing readers to consume the resulting messages as a stream.
+
+        The ``stream`` argument to the body modifies how the results are streamed back. Currently this
+        optional argument can be set to ``jsonlines`` to get newline separated JSON data.
+
+
+    *Examples*
+        The following two examples show querying the ``api/v1/storm`` endpoint and receiving multiple message types.
+
+        aiohttp example:
+
+        .. code:: python3
+            :name: aiohttp api/v1/storm example
+
+            import json
+            import pprint
+
+            # Assumes sess is an aiotthp client session that has previously logged in
+
+            query = '.created $lib.print($node.repr(".created")) | limit 3'
+            data = {'query': query, 'opts': {'repr': True}}
+            url = 'https://localhost:4443/api/v1/storm'
+
+            async with sess.get(url, json=data) as resp:
+                async for byts, x in resp.content.iter_chunks():
```

Comment:
should use the `chunkend` flag to reassemble the chunks in case they have been re-chunked by a proxy

Reply:
nevermind i see you've got that later